testplan/testing/multitest/driver/connection/base.py

```diff
@@ -0,0 +1,69 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Union
+from collections import defaultdict
+
+from testplan.common.utils.context import ContextValue
+
+
+class Direction(Enum):
+    CONNECTING = "connecting"
+    LISTENING = "listening"
+
+
+@dataclass
+class BaseConnectionInfo:
+    name: str  # name of the connection
+    service: str  # e.g. HTTP, TCP, FIX
+    protocol: str  # tcp, udp, file
+    identifier: Union[int, str, ContextValue]
```

Comment:
if `connection_rep` implemented for different Connection Info type, i don't think we still need this identifier